### PR TITLE
chore(babel-preset): create readme with basic docs 

### PR DESCRIPTION
**Pull request by husayt**
This would be very useful as is not covered anywhere. Will be also great to add to nuxt docs

## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Comment by Atinux** [comment]
Thanks @husayt !

Sounds good to me, maybe @manniL may look at some typos/formatting :)

File: packages/babel-preset-app/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      presets({ isServer }) {
+        return [
+          [
+            "@nuxt/babel-preset-app", options            
+          ]
+        ]
+      }
+    }
+```    
+where `options` is an object with paramaters of following type
```
Comment (aldarund) [inline packages/babel-preset-app/README.md]: *parameters

**Review by aldarund — CHANGES_REQUESTED** [review]

File: packages/babel-preset-app/README.md (hunk 1 above)
Comment (husayt) [reply packages/babel-preset-app/README.md]: thanks

File: packages/babel-preset-app/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# `@nuxt/babel-preset-app`
+## Default babel preset for nuxt
```
Comment (manniL) [inline packages/babel-preset-app/README.md]: Two headings directly underneath don't look decent IMO. Maybe

```suggestion
> Default babel preset for nuxt
```

File: packages/babel-preset-app/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+## Usage
+
+This is the default preset used by Nuxt, which is mainly a wrapper around  '@babel/preset-env' preset. It also optionally uses preset `@vue/babel-preset-jsx`, plugins `@babel/plugin-syntax-dynamic-import`, `@babel/plugin-proposal-decorators`, `@babel/plugin-proposal-class-properties`, `@babel/plugin-transform-runtime`  and adds polyfills.
```
Comment (manniL) [inline packages/babel-preset-app/README.md]:
```suggestion
This is the default preset used by Nuxt, which is mainly a wrapper around the `@babel/preset-env` preset. It also optionally uses the `@vue/babel-preset-jsx` preset as well as `@babel/plugin-syntax-dynamic-import`, `@babel/plugin-proposal-decorators`, `@babel/plugin-proposal-class-properties`, `@babel/plugin-transform-runtime`. Furthermore the preset is adding polyfills.
```

File: packages/babel-preset-app/README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+No additional configuration usually required. But there is an option to fine tune it's behavior if needed. Just add the following to nuxt-config
```
Comment (manniL) [inline packages/babel-preset-app/README.md]:
```suggestion
Usually, no additional configuration is required. If needed though, there is an option to fine-tune the preset's behavior. Just add the following to `nuxt.config.js`:
```

File: packages/babel-preset-app/README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+          ]
+        ]
+      }
+    }
+```    
+where `options` is an object with parameters of following type
+```
+ const options = {
+      useBuiltIns: "entry"
+      }
```
Comment (manniL) [inline packages/babel-preset-app/README.md]: Unformatted indents
```suggestion
}
```

File: packages/babel-preset-app/README.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+            "@nuxt/babel-preset-app", options            
+          ]
+        ]
+      }
+    }
+```    
+where `options` is an object with parameters of following type
+```
+ const options = {
+      useBuiltIns: "entry"
```
Comment (manniL) [inline packages/babel-preset-app/README.md]:
```suggestion
  useBuiltIns: "entry"
```

File: packages/babel-preset-app/README.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+      presets({ isServer }) {
+        return [
+          [
+            "@nuxt/babel-preset-app", options            
+          ]
+        ]
+      }
+    }
+```    
+where `options` is an object with parameters of following type
```
Comment (manniL) [inline packages/babel-preset-app/README.md]:
```suggestion
...where `options` is an object with parameters, for example:
```

File: packages/babel-preset-app/README.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+```
+ const options = {
+      useBuiltIns: "entry"
+      }
+```
+Below is the list of all available parameters
```
Comment (manniL) [inline packages/babel-preset-app/README.md]:
```suggestion
Below is a list of all available parameters:
```

File: packages/babel-preset-app/README.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+### Options
+* **buildTarget** passed by builder, either `"server"` or `"client"`
```
Comment (manniL) [inline packages/babel-preset-app/README.md]: Should we actually list it here if it's for internal usage only anyway?

```suggestion
* **buildTarget** - passed in through the Builder, either `"server"` or `"client"`
```

File: packages/babel-preset-app/README.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+* **configPath** - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#configpath)' parameter
```
Comment (manniL) [inline packages/babel-preset-app/README.md]: Please use this format where applicable:

```suggestion
* **configPath** - [`@babel/preset-env` parameter](https://babeljs.io/docs/en/babel-preset-env#configpath)
```

File: packages/babel-preset-app/README.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+* **shippedProposals** - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#shippedproposals)' parameter
+* **spec** - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#spec)' parameter
+* **targets** - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#targets)' parameter
+* **useBuiltIns**, default `"usage"` - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#usebuiltins)' parameter
+
+[Here](https://babeljs.io/docs/en/babel-preset-env#options) is a detailed docs for '@babel/preset-env' parameters
+
+### Example 1. Change targets for server and client respectively
+```js
+  babel: {
```
Comment (manniL) [inline packages/babel-preset-app/README.md]: Wrong indentation

File: packages/babel-preset-app/README.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+* **loose**, default `false` - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#loose)' parameter and also sets `loose=true` for `@babel/plugin-proposal-class-properties`
+* **modern** passed by builder, either `true` or `false`
+* **modules**, default `false` - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#modules)' parameter
+* **polyfills**, default `['es6.array.iterator','es6.promise','es7.promise.finally']`, more [here](https://github.com/zloirock/core-js)
+* **shippedProposals** - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#shippedproposals)' parameter
+* **spec** - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#spec)' parameter
+* **targets** - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#targets)' parameter
+* **useBuiltIns**, default `"usage"` - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#usebuiltins)' parameter
+
+[Here](https://babeljs.io/docs/en/babel-preset-env#options) is a detailed docs for '@babel/preset-env' parameters
```
Comment (manniL) [inline packages/babel-preset-app/README.md]: Not accessible :frowning: 

Never use [Here](...) as link.
```suggestion
There are [detailed docs](https://babeljs.io/docs/en/babel-preset-env#options) for the parameters of '@babel/preset-env'.
```

File: packages/babel-preset-app/README.md [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+* **decoratorsBeforeExport**
+* **decoratorsLegacy**, default true
+* **exclude** - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#exclude)' parameter
+* **ignoreBrowserslistConfig**, defaults to value of `modern` - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#ignorebrowserslistconfig)' parameter
+* **include** - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#include)' parameter
+* **jsx**, default truish, can be a an object passed as params to [@vue/babel-preset-jsx`](https://www.npmjs.com/package/@vue/babel-preset-jsx)
+* **loose**, default `false` - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#loose)' parameter and also sets `loose=true` for `@babel/plugin-proposal-class-properties`
+* **modern** passed by builder, either `true` or `false`
+* **modules**, default `false` - '[@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env#modules)' parameter
+* **polyfills**, default `['es6.array.iterator','es6.promise','es7.promise.finally']`, more [here](https://github.com/zloirock/core-js)
```
Comment (manniL) [inline packages/babel-preset-app/README.md]:
```suggestion
* **polyfills**, default `['es6.array.iterator','es6.promise','es7.promise.finally']`, more [in the corresponding repository](https://github.com/zloirock/core-js)
```

**Review by TheAlexLichter — CHANGES_REQUESTED** [review]

**Comment by manniL** [comment]
Thanks @husayt :relaxed: 
I left some comments.

File: packages/babel-preset-app/README.md (hunk 12 above)
Comment (husayt) [reply packages/babel-preset-app/README.md]: good point

File: packages/babel-preset-app/README.md (hunk 11 above)
Comment (husayt) [reply packages/babel-preset-app/README.md]: It's "prettier" in action

**Comment by husayt** [comment]
@manniL, thanks for excellent review.
Should be ok to merge now

**Review by aldarund — APPROVED** [review]

**Review by TheAlexLichter — APPROVED** [review]
Well done :)

**Pull request merged**